go/enclave/rollupchain/rollup_chain.go

```diff
@@ -295,18 +295,14 @@ func (rc *RollupChain) processState(rollup *obscurocore.Rollup, txs []*common.L2
 		if !f {
 			log.Panic("There should be an entry for each transaction ")
 		}
-		rec, ok := result.(*types.Receipt)
-		if ok {
+		rec, foundReceipt := result.(*types.Receipt)
+		if foundReceipt {
 			executedTransactions = append(executedTransactions, tx)
 			txReceipts = append(txReceipts, rec)
 		} else {
 			err := result.(error)
```

Comment:
I think you can get rid of this cast to error now, just put result in the log message below and let the `%s` stringer deal with it.